File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/core/models/base/Robot.java
package hu.elteik.knowledgelab.javaengine.core.models.base;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Robot {

    protected Long ID;
    protected RobotState state;
    protected Color color;
    protected Long onID;
    protected Long destinationID;

}

File: dispersion-app/src/app/models/base-entities/Robot.ts
export enum RobotState {
  START = 'START',
  EXPLORE = 'EXPLORE',
  LEADER = 'LEADER',
  SETTLED = 'SETTLED',
  TERMINATED = 'TERMINATED'
}

export class Robot {

  id: number;
  onID: number;
  state: RobotState;
  color: string;
  destinationID: number;

  constructor(id?: number, onID?: number, state?: RobotState, color?: string, destinationID?: number) {
    this.id = id;
    this.onID = onID;
    this.state = state;
    this.color = color;
    this.destinationID = destinationID;
  }

}
